torch/fx/passes/tensorify_python_scalars.py

```diff
@@ -0,0 +1,241 @@
+
+        # hash cons on arguments, run expr handler
+        expr_to_tensor_proxy[expr] = _run_sympy_handler(
+            Analysis,
+            [_sympy_interp(arg) for arg in expr.args],  # type: ignore[arg-type]
+            expr,
+        )
+
+        return expr_to_tensor_proxy[expr]
+
+    nodes = list(graph.nodes)
+    for i, node in enumerate(nodes[:-1]):
+        with graph.inserting_before(
+            nodes[i + 1] if node not in placeholders else first_non_placeholder
+        ):
+            # Look for tensor.item() calls on placeholders
+            if unbacked_bindings := node.meta.get("unbacked_bindings"):
+                for s in unbacked_bindings.keys():
+                    if (
+                        node is not None
+                        and node.op == "call_function"
+                        and node.target is torch.ops.aten._local_scalar_dense.default
+                    ):
+                        dtype = node.args[0].meta["val"].dtype
+                        if dtype != torch.float64:
+                            continue
+
+                        assert isinstance(node.args[0], fx.Node), node.args[0]
+
+                        expr_to_tensor_proxy[s] = fx.Proxy(node.args[0], tracer=tracer)
+                        expr_to_sym_proxy[s] = fx.Proxy(node, tracer=tracer)
+
+            elif (sym_expr := _get_sym_val(node)) is not None:
+                if sym_expr not in expr_to_sym_proxy and not isinstance(
+                    sym_expr, (sympy.Number, sympy.logic.boolalg.BooleanAtom)
+                ):
+                    expr_to_sym_proxy[sym_expr] = fx.Proxy(node, tracer=tracer)
+
+            # Look for functions to convert
+            if node.op == "call_function" and node.target is torch.ops.aten.mul.Tensor:
+                args = []
+                transform = False
+                compute_dtype, result_dtype = elementwise_dtypes(
+                    node.meta["val"],
+                    type_promotion_kind=ELEMENTWISE_TYPE_PROMOTION_KIND.DEFAULT,
+                )
+
+                for a in node.args:
+                    if isinstance(a, fx.Node) and isinstance(
+                        zf := a.meta["val"], torch.SymFloat
+                    ):
+                        transform = True
+                        try:
+                            a = _sympy_interp(zf.node.expr).node
+                        except NotImplementedError:
+                            transform = False
+                            break
+
+                        if "val" not in a.meta or a.meta["val"].dtype != compute_dtype:
+                            res = graph.call_function(
+                                torch.ops.prims.convert_element_type.default,
+                                (
+                                    a,
+                                    compute_dtype,
+                                ),
+                            )
+                            res.meta = node.meta
```

Comment:
This passes all the tests but feels weird to do. Ideally I'd do something like

```
res.meta[
    "val"
] = torch.ops.prims.convert_element_type.default(
    a.meta["val"], compute_dtype
)
```

but we don't always have res.meta["val"].  For example in `python test/inductor/test_torchinductor.py CpuTests.test_pow_symfloat_cpu` we don't. I'm still digesting metas/vals/fake tensors/etc. so if any of the reviewers have a better way to populate meta here, I'm all ears!